custom_components/oasis_mini/pyoasiscontrol/device.py

```
@@ -0,0 +1,762 @@
+        cloud: OasisCloudClient | None = None,
+        client: OasisClientProtocol | None = None,
+    ) -> None:
+        # Transport
+        """
+        Initialize an OasisDevice with identification, network, transport references, and default state fields.
+        
+        Parameters:
+            model (str | None): Device model identifier.
+            serial_number (str | None): Device serial number.
+            name (str | None): Human-readable device name; if omitted, defaults to "<model> <serial_number>".
+            ssid (str | None): Last-known Wi‑Fi SSID for the device.
```

Ruff (0.14.5)

Docstring contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)